src/schemathesis/loaders.py

```diff
@@ -54,13 +55,18 @@ def from_uri(
     endpoint: Optional[Filter] = None,
     tag: Optional[Filter] = None,
     operation_id: Optional[Filter] = None,
+    port: Optional[int] = None,
     *,
     app: Any = None,
     validate_schema: bool = True,
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
     """Load a remote resource and parse to schema instance."""
     kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if not base_url and port:
+        o = urlparse(uri)
+        loc = re.sub(r':(\d+)$', '', o.netloc)
```

Comment:
The `ParseResult` class provides `hostname` and `port` - I suggest using them instead of regex (it will require catching `ValueError` in some corner case).

For example, an URL like this one will be mangled - "http://2001:0db8:85a3:0000:0000:8a2e:0370:7334/openapi.json".
The base URL will end up with a different IPv6 address. E.g. with port 8081 it will be "2001:0db8:85a3:0000:0000:8a2e:0370:8081" instead of "[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:8081"

I also suggest adding handling & a test case for IPv6 addresses.

Reply:
Thanks - that's a good idea, didn't think of ipv6